examples/vault-auto-unseal/main.tf

```diff
@@ -0,0 +1,163 @@
+# ---------------------------------------------------------------------------------------------------------------------
+# DEPLOY A VAULT SERVER CLUSTER AND A CONSUL SERVER CLUSTER IN AWS
+# This is an example of how to launch a vault cluster and then authenticate an instance to the cluster
+# ---------------------------------------------------------------------------------------------------------------------
+
+terraform {
+  required_version = ">= 0.11.0"
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# Creates a KMS key that will be used to encrypt and decrypt Vault's master key
+# ---------------------------------------------------------------------------------------------------------------------
+
+resource "aws_kms_key" "vault" {
+  description             = "Vault unseal key"
+  deletion_window_in_days = 7
+
+  tags {
+    Name = "vault-unseal-example"
+  }
+}
```

Comment:
Hm... I don't think we should create a new CMK here every time. Each one costs $1/month, even if you delete it immediately. If we have hundreds of test runs per month, that starts to get expensive.

Instead, you can take in an alias as a parameter, and look up an existing key using the [aws_kms_key data source](https://www.terraform.io/docs/providers/aws/d/kms_key.html). IIRC, in our test account, we already have  an existing CMK we can use in every region, but you may want to check that.

Reply:
+1 to this comment. I believe I manually created a KMS key in every AWS region at one point. Of course, some new AWS regions have come out since then so it's possible there are a few missing here or there.

Reply:
okay!